errors.go

```diff
@@ -56,10 +56,11 @@ const (
 //         Err: err,
 //     }.
 type Error struct {
-	Code string
-	Msg  string
-	Op   string
-	Err  error
+	Code              string
+	Msg               string
+	Op                string
+	Err               error
+	ValidationErrBody map[string]interface{}
```

Comment:
I would just use the Err field. implement an ValidationErr 
```go
     type ValidationError struct{...}

    func (e *ValidationError) Error() string{}
```
So the error will become
```go
     &Error{
        Code: ...,
        Msg: ...
        Err: &ValidationError{...}
    }
```

Reply:
hmm, the issue with that is I want the error to be a JSON body only for the HTTP API, in the CLI I just want log format, which is what the Error() method is providing now. There is nothing stopping a user from providing an Err in addition to the validation error as is:

```go
&Error{
        Code: ...,
        Msg: ...
        Err: FOO_ERROR,
        ValidationErrs: // detailed view of the error mentioned above
}
```
The biggest concern for me is it becomes a different experience for the same field in 1 place and not others.

Reply:
The err field is initially designed for different kinds of complex error.  One reason to hide that field is to prevent reading difficulties. 

For example, the flux parsing error is now, instead of a complex embedding error.
```
{
  "code": "internal error",
  "message": "compilation failed: loc 2:17-2:23: expected an operator between two expressions"
}
```

Since your error won't be fit in a simple err message, feels like it's wiser to bring that field back, instead of adding new one. Especially, this is only the pkgr error, not really a domain level error.